addons/purchase/models/purchase.py

```diff
@@ -1188,7 +1188,7 @@ def _prepare_add_missing_fields(self, values):
 
     @api.model
     def _prepare_purchase_order_line(self, product_id, product_qty, product_uom, company_id, supplier, po):
-        partner = supplier.name
+        partner = supplier
```

Comment:
supplier is 'product.supplierinfo', and name is a many2one field of partner.
https://github.com/odoo/odoo/blob/14.0/addons/product/models/product.py#L732